pkg/workload/schemachange/schemachange.go

```diff
@@ -573,6 +582,30 @@ func (l *logger) addExpectedErrors(execErrors errorCodeSet, commitErrors errorCo
 	}
 }
 
+// flushLogWithError outputs the currentLogEntry of the schemaChangeWorker, with
+// an error message (any available error state information is also added).
+// It is a noop if l.verbose < 0.
+func (l *logger) flushLogWithError(tx pgx.Tx, err error) {
+	if l.verbose < 1 {
+		return
+	}
+
+	// Fetch and apply the error state to the log entry.
+	func() {
+		l.currentLogEntry.mu.Lock()
+		defer l.currentLogEntry.mu.Unlock()
+		if l.currentLogEntry.mu.entry != nil {
+			var state *ErrorState
+			if errors.As(err, &state) {
+				l.currentLogEntry.mu.entry.ErrorState = state
+			}
+		}
+	}()
+
+	l.flushLogAndLock(tx, err.Error(), true)
+	l.currentLogEntry.mu.Unlock()
```

Comment:
Is this intentional? Looks like you're unlocking twice.

Reply:
Yes, the flushLogAndLock holds onto the lock. Not a fan of this design, I'll poke at it with a later PR.